packages/graph/hash_graph/lib/graph/src/store/postgres/knowledge/entity/mod.rs

```diff
@@ -372,45 +385,82 @@ impl<C: AsClient> PostgresStore<C> {
 impl<C: AsClient> EntityStore for PostgresStore<C> {
     async fn create_entity(
         &mut self,
-        properties: EntityProperties,
-        entity_type_id: VersionedUri,
-        owned_by_id: OwnedById,
-        entity_uuid: Option<EntityUuid>,
+        entity_id: EntityId,
+        decision_time: Option<Timestamp>,
         updated_by_id: UpdatedById,
+        archived: bool,
+        entity_type_id: VersionedUri,
+        properties: EntityProperties,
         link_data: Option<LinkData>,
     ) -> Result<EntityMetadata, InsertionError> {
-        let transaction = PostgresStore::new(
-            self.as_mut_client()
-                .transaction()
-                .await
-                .into_report()
-                .change_context(InsertionError)?,
-        );
+        let entity_type_version_id = self
+            .version_id_by_uri(&entity_type_id)
+            .await
+            .change_context(InsertionError)?;
 
-        let entity_uuid = entity_uuid.unwrap_or_else(|| EntityUuid::new(Uuid::new_v4()));
-        let entity_id = EntityId::new(owned_by_id, entity_uuid);
+        let properties = serde_json::to_value(properties)
+            .into_report()
+            .change_context(InsertionError)?;
 
-        // TODO: match on and return the relevant error
-        //   https://app.asana.com/0/1200211978612931/1202574350052904/f
-        transaction.insert_entity_uuid(entity_uuid).await?;
-        let metadata = transaction
-            .insert_entity(
-                entity_id,
-                properties,
-                entity_type_id,
-                updated_by_id,
-                link_data,
+        let row = self
+            .as_client()
+            .query_one(
+                r#"
+                SELECT
+                    entity_edition_id,
+                    lower(decision_time),
```

Comment:
Are we storing them as strings in the database?

Reply:
if you're asking about the `lower` it's the way to get the lower bound, not the lowercase version of the string

Reply:
https://www.w3resource.com/PostgreSQL/lower-function.php

I was misled 😭

Reply:
If this is a custom function, do we want to pick something that doesn't clash with the in-built? And if it is the in-built... then what the heck

Reply:
The built-in functions are called `lower` and `upper`.